x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/analyses/Classification.java

```diff
@@ -0,0 +1,190 @@
+    public static final ParseField PREDICTION_FIELD_NAME = new ParseField("prediction_field_name");
+    public static final ParseField NUM_TOP_CLASSES = new ParseField("num_top_classes");
+    public static final ParseField TRAINING_PERCENT = new ParseField("training_percent");
+
+    private static final ConstructingObjectParser<Classification, Void> LENIENT_PARSER = createParser(true);
+    private static final ConstructingObjectParser<Classification, Void> STRICT_PARSER = createParser(false);
+
+    private static ConstructingObjectParser<Classification, Void> createParser(boolean lenient) {
+        ConstructingObjectParser<Classification, Void> parser = new ConstructingObjectParser<>(
+            NAME.getPreferredName(),
+            lenient,
+            a -> new Classification(
+                (String) a[0],
+                new BoostedTreeParams((Double) a[1], (Double) a[2], (Double) a[3], (Integer) a[4], (Double) a[5]),
+                (String) a[6],
+                (Integer) a[7],
+                (Double) a[8]));
+        parser.declareString(constructorArg(), DEPENDENT_VARIABLE);
+        BoostedTreeParams.declareFields(parser);
+        parser.declareString(optionalConstructorArg(), PREDICTION_FIELD_NAME);
+        parser.declareInt(optionalConstructorArg(), NUM_TOP_CLASSES);
+        parser.declareDouble(optionalConstructorArg(), TRAINING_PERCENT);
+        return parser;
+    }
+
+    public static Classification fromXContent(XContentParser parser, boolean ignoreUnknownFields) {
+        return ignoreUnknownFields ? LENIENT_PARSER.apply(parser, null) : STRICT_PARSER.apply(parser, null);
+    }
+
+    private final String dependentVariable;
+    private final BoostedTreeParams boostedTreeParams;
+    private final String predictionFieldName;
+    private final Integer numTopClasses;
+    private final double trainingPercent;
+
+    public Classification(String dependentVariable,
+                          BoostedTreeParams boostedTreeParams,
+                          @Nullable String predictionFieldName,
+                          @Nullable Integer numTopClasses,
+                          @Nullable Double trainingPercent) {
+        if (numTopClasses != null && (numTopClasses < 0 || numTopClasses > 1000)) {
+            throw ExceptionsHelper.badRequestException("[{}] must be an integer in [0, 1000]", NUM_TOP_CLASSES.getPreferredName());
+        }
+        if (trainingPercent != null && (trainingPercent < 1.0 || trainingPercent > 100.0)) {
+            throw ExceptionsHelper.badRequestException("[{}] must be a double in [1, 100]", TRAINING_PERCENT.getPreferredName());
+        }
+        this.dependentVariable = ExceptionsHelper.requireNonNull(dependentVariable, DEPENDENT_VARIABLE);
+        this.boostedTreeParams = ExceptionsHelper.requireNonNull(boostedTreeParams, BoostedTreeParams.NAME);
+        this.predictionFieldName = predictionFieldName;
+        this.numTopClasses = numTopClasses;
```

Comment:
Does `num_top_classes` have a fixed default value? If so we should set it explicitly.

Reply:
Done.

I think the default value should be "0".